packages/ti_google_threat_intelligence/elasticsearch/transform/ip_ioc/transform.yml

```
@@ -0,0 +1,45 @@
+# Use of "*" to use all namespaces defined.
+source:
+  index:
+    - "logs-ti_google_threat_intelligence.*"
+  query:
+    bool:
+      filter:
+        - match_phrase:
+            threat.indicator.type: ipv4-addr
+      must_not:
+        - exists:
+            field: error.message
+dest:
+  index: "logs-ti_google_threat_intelligence_latest.dest_ip_ioc-1"
+  aliases:
+    - alias: "logs-ti_google_threat_intelligence_latest.ip_ioc"
+      move_on_creation: true
+latest:
+  unique_key:
+    - event.dataset
+    - threat.indicator.ip
+  sort: "@timestamp"
+description: >-
+  Latest IP Address IOC from Google Threat Intelligence API. As IP Address IOC get updated, this transform stores only the latest state of each IP Address IOC inside the destination index. Thus the transform's destination index contains only the latest state of the IP Address IOC.
+frequency: 30s
+settings:
+  # This is required to prevent the transform from clobbering the Fleet-managed mappings.
+  deduce_mappings: false
+  unattended: true
+sync:
+  time:
+    field: "event.ingested"
+    # Updated to 120s because of refresh delay in Serverless. With default 60s,
+    # sometimes transform wouldn't process all documents.
+    delay: 120s
+retention_policy:
+  time:
+    field: '@timestamp'
+    max_age: 30d
+_meta:
+  managed: false
+  # Bump this version to delete, reinstall, and restart the transform during
+  # package installation.
+  fleet_transform_version: 1.0.0
+  run_as_kibana_system: false
```

```suggestion
  run_as_kibana_system: false
```
Remove this, it shouldn't be needed for TI* transforms, they have permissions to run as kibana_system role.

The new `rule` transform may still need `run_as_kibana_system: false`, because its reading for `.internal*` indices

---

Hello @kcreddy,
Please Find the Drive Link: https://drive.google.com/file/d/1_Z8kbECp890fuT2ULQtxyI_0Do4qEAyH/view?usp=drive_link

---

Thanks @niraj-crest, really appreciate!

Can you please check if it resolves by changing the source indices to match this generic pattern: `logs-ti_*.*-*`?
Possibly:
```yaml
source:
  index:
    - "logs-ti_google_threat_intelligence.*-*"
```

---

Yes the `rule` transform still requires `run_as_kibana_system: false`
https://github.com/elastic/integrations/pull/13189#discussion_r2112389395

---

By following the suggested pattern and removing `run_as_kibana_system` from non-rule transforms, the reauthorization warning issue has been resolved.

However, there's an important consideration here. As discussed in [this comment](https://github.com/elastic/integrations/pull/13189/files/595a25b8cf2f290ab278e11d05c4d38e13fd5986#r2142637521), this PR is part of the Google Threat Intelligence integration and this integration will includes both the threat list and IOC stream along with their assets. Therefore, to make the transform logic more robust and avoid processing enriched logs or errors, the following query has been added:
```yml
source:
  index:
    - "logs-ti_google_threat_intelligence.*-*"
  query:
    bool:
      filter:
        - match_phrase:
            threat.indicator.type: domain-name
        - match_phrase:
            event.dataset: ti_google_threat_intelligence.ioc_stream            
      must_not:
        - exists:
            field: error.message
        - match_phrase:
            event.dataset: ti_google_threat_intelligence.enriched_ioc
        - match_phrase:
            event.dataset: ti_google_threat_intelligence.enriched_ioc_stream

```

This works well in the live environment.
However, system tests are failing, likely because query conditions are not applied or evaluated during the system test.

Please find example of system test result for `iot` data-stream after configuring suggested changes.

```
--- Test results for package: ti_google_threat_intelligence - START ---
╭───────────────────────────────┬─────────────┬───────────┬───────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─────────────────╮
│ PACKAGE                       │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT                                                                                                              │    TIME ELAPSED │
├───────────────────────────────┼─────────────┼───────────┼───────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────────────┤
│ ti_google_threat_intelligence │ iot         │ system    │ default   │ ERROR: no documents found in preview for transform "logs-ti_google_threat_intelligence.domain_ioc_st-default-0.1.0" │ 2m10.257972784s │
╰───────────────────────────────┴─────────────┴───────────┴───────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────╯
--- Test results for package: ti_google_threat_intelligence - END   ---
Done
Error: failed to process results: one or more test cases failed
```

---

Do you mean the system tests were working fine when `run_as_kibana_system: false` but not otherwise?

> However, system tests are failing, likely because query conditions are not applied or evaluated during the system test.

This needs to be debugged further. Please open an issue in https://github.com/elastic/elastic-package with screenshots or recording.
To unblock you, add a link to that issue in this transform file as a comment indicating the reason for `run_as_kibana_system: false`.

---

> Would you recommend we open an issue in the [elastic-package](https://github.com/elastic/elastic-package) regarding this?

@niraj-crest, yes please create the issue there with all details. Especially confirming that you are receiving data in the destination indices correctly, but the system tests are failing with `ERROR: no documents found`

---

I have updated the PR. Could you please review the changes? If everything looks good, we can proceed with merging the PR.

---

Hello @kcreddy,
We have added suggested comment can you please review it.

---

This should be done before making the integration GA.
When you have an outstanding list of pending items (that we discussed), please create a public issue and link to this PR. It will be the items required to make the integration GA. For example: https://github.com/elastic/integrations/pull/13189#discussion_r2135449116

---

hello @kcreddy,
As suggested we have raised 2 issues one for https://github.com/elastic/integrations/pull/13189#discussion_r2135449116 and one for making package GA.

- https://github.com/elastic/integrations/issues/14286
- https://github.com/elastic/integrations/issues/14287

We will appreciate if you approve and merge this PR.